doc/specs/#885 - winrt Terminal Settings.md

```diff
@@ -0,0 +1,211 @@
+
+    // Create a copy of the existing AppSettings
+    AppSettings Clone();
+    
+    // Compares object to "source" and applies changes to
+    // the settings file at "outPath"
+    void Save(String outPath);
+}
+```
+
+#### TerminalApp: Loading and Reloading Changes
+
+TerminalApp will construct and reference an `AppSettings settings` as follows:
+- TerminalApp will have a global reference to "defaults.json" and "settings.json" filepaths
+- construct an `AppSettings` using the default constructor
+- `settings.LayerSettings("defaults.json")`
+- `settings.LayerSettings("settings.json")`
+
+**NOTE:** This model allows us to layer even more settings files on top of the existing Terminal Settings
+ Model, if so desired. This could be helpful when importing additional settings files from an external location
+ such as a marketplace.
+
+When TerminalApp detects a change to "settings.json", it'll repeat the steps above. We could cache the result from
+ `settings.LayerSettings("defaults.json")` to improve performance.
+
+Additionally, TerminalApp would reference `settings` for the following functionality...
+- `Profiles` to populate the dropdown menu
+- `Keybindings` to detect active key bindings and which actions they run
+- `Commands` to populate the Command Palette
+- `Warnings` to show a serialization warning, if any
+
+#### TerminalControl: Acquiring and Applying the Settings
+
+At the time of writing this spec, TerminalApp constructs `TerminalControl.TerminalSettings` WinRT objects
+ to expose `IControlSettings` and `ICoreSettings` to any hosted terminals. In moving `IControlSettings`
+ and `ICoreSettings` down to the TerminalControl layer, TerminalApp can now have better control over
+ how to expose relevant settings to a TerminalControl instance.
+
+`TerminalSettings` (which implements `IControlSettings` and `ICoreSettings`) will be moved to TerminalApp and act as a bridge that queries `AppSettings`. Thus,
+ when TermControl requests `IControlSettings` data, `TerminalSettings` responds by extracting the
+ relevant information from `AppSettings`. TerminalCore operates the same way with `ICoreSettings`.
+
+#### Settings UI: Modifying and Applying the Settings
+
+The Settings UI will also have a reference to the `AppSettings settings` from TerminalApp
+ as `settingsSource`. When the Settings UI is opened up, the Settings UI will also have its own `AppSettings settingsClone`
+ that is a clone of TerminalApp's `AppSettings`.
+```c++
+settingsClone = settingsSource.Clone()
+```
+
+As the user navigates the Settings UI, the relevant contents of `settingsClone` will be retrieved and presented.
+ As the user makes changes to the Settings UI, XAML will update `settingsClone` using XAML data binding.
+ When the user saves/applies the changes in the XAML, `settingsClone.Save("settings.json")` is called; 
+ this compares the changes between `settingsClone` and `settingsSource`, then injects the changes (if any) to `settings.json`.

```

Comment:
Maybe add a comment here that we're comparing changes between the modified settings in `settingsClone` and `settingsSource`, so that if the user has multiple settings files (#2933), we're persisting the changes to the root settings file, to try and reduce confusion.

That also might introduce more confusion - maybe best left for future considerations

Reply:
Moved to future considerations. This will be covered in the addendum though. :)